kasa/iot/modules/dimmer.py

```diff
@@ -0,0 +1,245 @@
+                type=Feature.Type.Switch,
+                category=Feature.Category.Config,
+            )
+        )
+
+        self._add_feature(
+            Feature(
+                device=self._device,
+                container=self,
+                id="dimmer_fade_on_time",
+                name="Dimmer Fade On Time",
+                icon="mdi:clock-out",
+                attribute_getter="fade_on_time",
+                attribute_setter="set_fade_on_time",
+                type=Feature.Type.Switch,
+                category=Feature.Category.Config,
+            )
+        )
+
+        self._add_feature(
+            Feature(
+                device=self._device,
+                container=self,
+                id="dimmer_gentle_off_time",
+                name="Dimmer Gentle Off Time",
+                icon="mdi:clock-in",
+                attribute_getter="gentle_off_time",
+                attribute_setter="set_gentle_off_time",
+                type=Feature.Type.Switch,
+                category=Feature.Category.Config,
+            )
+        )
+
+        self._add_feature(
+            Feature(
+                device=self._device,
+                container=self,
+                id="dimmer_gentle_on_time",
+                name="Dimmer Gentle On Time",
+                icon="mdi:clock-out",
+                attribute_getter="gentle_on_time",
+                attribute_setter="set_gentle_on_time",
+                type=Feature.Type.Switch,
+                category=Feature.Category.Config,
+            )
+        )
+
+        self._add_feature(
+            Feature(
+                device=self._device,
+                container=self,
+                id="dimmer_ramp_rate",
+                name="Dimmer Ramp Rate",
+                icon="mdi:clock-fast",
+                attribute_getter="ramp_rate",
+                attribute_setter="set_ramp_rate",
+                type=Feature.Type.Switch,
+                category=Feature.Category.Config,
+            )
+        )
+
+    def query(self) -> dict:
+        """Request Dimming configuration."""
+        req = merge(
+            self.query_for_command("get_dimmer_parameters"),
+            self.query_for_command("get_default_behavior"),
+        )
+
+        return req
+
+    @property
+    def config(self) -> dict:
+        """Return current configuration."""
+        return self.data["get_dimmer_parameters"]
+
+    @property
+    def threshold_min(self) -> int | None:
+        """Return the minimum dimming level for this dimmer."""
+        if (min := self.config.get("minThreshold")) is not None:
+            return int(min)
+        return None
+
+    async def set_threshold_min(self, min: int) -> dict:
+        """
+        Set the minimum dimming level for this dimmer.
+
+        The value will depend on the luminaries connected to the dimmer.
+
+        :param min: The minimum dimming level, in the range 0-51.
+        """
+        if (min < 0) or (min > 51):
```

Comment:
```suggestion
        if min < 0 or min > 51:
```
Is it really up to 51 and not 50? In any case, no need for parentheses here.

Reply:
Yeah if you use the app to set the dimmer trimto max, it sets the value to 51, which surprised me as well. The API rejects 52 as a value though, so it has to end on 51. Perhaps there is a special bypass mode on the triac or something for no dimming, and that's a special value used to denote that?

Reply:
Could be something special indeed, but we can keep it as it is and adjust when someone chimes in with an issue report :)

Reply:
Could be worth adding a small comment here about the tested app behaviour.